modules/setting/setting.go

```diff
@@ -230,11 +230,14 @@ func LoadSettingsForInstall() {
 	loadMailerFrom(CfgProvider)
 }
 
-var uniquePaths = make(map[string]string)
+var configuredPaths = make(map[string]string)
 
-func fatalDuplicatedPath(name, p string) {
-	if targetName, ok := uniquePaths[p]; ok && targetName != name {
-		log.Fatal("storage path %q is being used by %q and %q and all storage paths must be unique to prevent data loss.", p, targetName, name)
+func checkOverlappedPath(name, path string) {
+	// TODO: some paths shouldn't overlap (storage.xxx.path), while some could (data path is the base path for storage path)
+	if targetName, ok := configuredPaths[path]; ok && targetName != name {
+		msg := fmt.Sprintf("Configured path %q is used by %q and %q at the same time. The paths must be unique to prevent data loss.", path, targetName, name)
+		log.Error("%s", msg)
+		DeprecatedWarnings = append(DeprecatedWarnings, msg)
 	}
```

Comment:
Can't we fix this completely using
```suggestion
	path = gopath.Clean(path)
	
	for existingPath, existingName := range configuredPaths {
		if strings.HasPrefix(path, existingPath) || strings.HasPrefix(existingPath, path) {
			var msg string
			if path == existingPath {
				msg = fmt.Sprintf("Path %q is used by %q and %q simultaneously. This may lead to data loss. To prevent this, the paths must be different and they must not overlap.", path, existingName, name)
			} else {
				msg = fmt.Sprintf("Path %q (used by %q) is contained within %q (used by %q). This may lead to data loss. To prevent this, the paths may not overlap.", existingPath, existingName, path, name)
			}
		log.Error("%s", msg)
		DeprecatedWarnings = append(DeprecatedWarnings, msg)
	}
```
(assuming we have an `import gopath path` above)
?

Reply:
It is not right (incomplete_. `/a/b` overlaps `/a`, meanwhile `/a` overlaps` /a/b`

If you would like to "complete" it, it also needs enough tests.

Reply:
Yup, fixed my comment.

Reply:
But your "msg" is still not right, and it doesn't have tests.

As I said: "I won't do any further change". So feel free to take it, or improve it, or discard it.

Reply:
And one more thing, `path = gopath.Clean(path)` is also not right. It should use `filepath`.